### PR TITLE
CGAL:: Use std atomic and threads

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Deprecate `cpp11::atomic` and `cpp11::threads` and use std versions instead

## Release Management

* Affected package(s):Installation

### CODE REVIEW

File: Mesh_3/include/CGAL/Mesh_vertex_base_3.h [hunk 1]
```diff
@@ -81,7 +81,7 @@ class Mesh_vertex_base_3_base<Parallel_tag>
   }
 
 protected:
-  typedef tbb::atomic<unsigned int> Erase_counter_type;
+  typedef std::atomic<unsigned int> Erase_counter_type;
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_vertex_base_3.h:84]: Should not you include `<atomic>`?

File: Mesh_3/include/CGAL/Mesh_vertex_base_3.h [hunk 2]
```diff
@@ -66,6 +66,13 @@ template <>
 class Mesh_vertex_base_3_base<Parallel_tag>
 {
 public:
+  Mesh_vertex_base_3_base()
+  {}
+
+  Mesh_vertex_base_3_base( const Mesh_vertex_base_3_base& c)
+  {
+    m_erase_counter.store(c.erase_counter());
+  }
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_vertex_base_3.h:75]: I see that this copy-constructor is needed by Mesh_3 global optimizers. Where as the `m_erase_counter` is actually only required by the core of the meshing process, in parallel.

I really wish we had dynamic properties, in CGAL triangulations, like in `CGAL::Surface_mesh`!

**Comment by maxGimeno** [comment]
error : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-125/Polyhedron_Demo/TestReport_lrineau_Fedora.gz

**Comment by maxGimeno** [comment]
errors: https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-128/Classification_Examples/TestReport_lrineau_Ubuntu-Latest-NO_DEPRECATED_CODE.gz

**Comment by lrineau** [comment]
#5687 makes CGAL able to find oneAPI-TBB (using the `CONFIG` mode of CMake `find_package`), but `tbb::atomic` does not exist, in oneAPI-TBB. So now this PR #5692 is mandatory to build CGAL using TBB, if both TBB-2020.1 and oneAPI-TBB-2021 are installed in parallel (like it is on my machine).

With this PR, that seems to pass.

**Comment by maxGimeno** [comment]
Successfully tested on https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-129.shtml

**Pull request merged**